solr/core/src/java/org/apache/solr/metrics/prometheus/core/SolrCoreMetric.java

```diff
@@ -25,28 +26,26 @@
 
 /** Base class is a wrapper to export a solr.core {@link com.codahale.metrics.Metric} */
 public abstract class SolrCoreMetric extends SolrMetric {
+
   public String coreName;
 
-  public SolrCoreMetric(
-      Metric dropwizardMetric, String coreName, String metricName, boolean cloudMode) {
+  public SolrCoreMetric(Metric dropwizardMetric, String metricName) {
     super(dropwizardMetric, metricName);
-    this.coreName = coreName;
-    labels.put("core", coreName);
-    if (cloudMode) {
-      appendCloudModeLabels();
-    }
-  }
-
-  private void appendCloudModeLabels() {
-    Matcher m = CLOUD_CORE_PATTERN.matcher(coreName);
-    if (m.find()) {
-      labels.put("collection", m.group(1));
-      labels.put("shard", m.group(2));
-      labels.put("replica", m.group(3));
+    Matcher cloudPattern = CLOUD_CORE_PATTERN.matcher(metricName);
+    Matcher standalonePattern = STANDALONE_CORE_PATTERN.matcher(metricName);
+    if (cloudPattern.find()) {
+      coreName = cloudPattern.group(1);
+      labels.put("core", cloudPattern.group(1));
+      labels.put("collection", cloudPattern.group(2));
+      labels.put("shard", cloudPattern.group(3));
+      labels.put("replica", cloudPattern.group(4));
```

Comment:
I believe it's possible to have named groups in a regular expression.  If you enhance your regular expressions as such, you could generically apply the pattern with Matcher and extract all named groups without hard-coding the expectations separate from the regular expression, which is annoying and error-prone to maintain.

Reply:
Cool, learned something new. Added.